dask/dataframe/groupby.py

```diff
@@ -341,6 +341,76 @@ def _groupby_aggregate(
     return aggfunc(grouped, **kwargs)
 
 
+def _groupby_aggregate_spec(
```

Comment:
Is this just for ergonomics, or does the existing `_groupby_aggregate` function not fully have the functionality we need?

Reply:
This was mostly for ergonomics. A previous version of this just fed `spec` into `_groupby_aggregate` as an optional kwarg, but I didn't really like having that function have two different behaviors based on proliferating kwargs, so I split it out. Mostly a gut feeling, rather than any strong reasoning.